### PR TITLE
CI: Make a useful migration test

### PR DESCRIPTION
**Pull request by efritz**
This PR was originally meant to make the CI check for backwards/forwards runnable migrations more useful (by removing a piece of easy code that prevented us from seeing a bunch of problems).

This revealed a few such problems! I've updated the migrations that can't be ran twice so that the test will initially pass. I've also discovered an off-by-one error in the downgrade path for the new migration code, fixed here.

### CODE REVIEW

File: migrations/frontend/1528395880_insights_queue_persist_mode.down.sql [hunk 1]
```diff
@@ -12,3 +12,5 @@ ALTER TABLE insights_query_runner_jobs
     DROP COLUMN persist_mode;
 
 COMMIT;
+
+DROP TYPE PersistMode;
```
Comment (eseliger) [inline migrations/frontend/1528395880_insights_queue_persist_mode.down.sql:16]: Is it not possible to drop types in txns or why is this here?

**Review by eseliger — APPROVED** [review]
Good stuff! I think ideally we would also compare the schema after the up->down->up to a fresh up'ed DB in that test.

**Review by mrnugget — APPROVED** [review]
